_posts/2022-03-22-publiccode-yaml-analyse.md

```diff
@@ -0,0 +1,106 @@
+
+Weitere Informationen zum Publiccode.yml-Standard finden sich [auf der offiziellen italienischen Seite zum Publiccode-Standard](https://docs.italia.it/italia/developers-italia/publiccodeyml-en/en/master/index.html).
+
+Das Ergebnis ist in folgender Tabelle zu sehen:
+
+## Übersicht der Repositories
+
+ <iframe width="100%" height="500" src="https://tomsrocket.github.io/github-publiccode/public/" title="OSS Developer Community Health Analysis for Publiccode Github Repositories"></iframe>
+Quelle: [https://tomsrocket.github.io/github-publiccode/public/](https://tomsrocket.github.io/github-publiccode/public/)
+
+<br />
+Spalte 1 enthält einen Farbcode nach Ampel-Schema. Dadurch wird versucht, die "Qualität" der jeweiligen Entwicklercommunity grob abzuschätzen.
+Die Spalten 3-6 der Tabelle verdeutlichen das Bewertungsschema, das der Abschätzung zu Grunde liegt:
+
+## Versuch einer Bewertung der Repositories
+
+Über die Github-API können für jedes Repository einige interessante Werte abgerufen werden:
+
+* Wann war der letzte Commit?
+* Wieviele Commits gab es bisher insgesamt?
+* Wieviele unterschiedliche Committer, bzw. "Contributors" hat das Repository?
+* Wieviele Stars hat das Repository auf Github?
+
+Anhand dieser vier Kennzahlen lässt sich schon relativ gut abschätzen, ob hinter dem Repository eine lebendige Community steht, und ob das Projekt noch aktiv weiterentwickelt wird:
+* **Letzter Commit mehrere Jahre alt** => Projekt ist nicht mehr in aktiver Entwicklung
+* **Nur sehr wenige Commits** => Risiko eines "Fake"-Open-Source-Projekts, ohne richtige Community dahinter. Z.B.: Eine Firma entwickelt es intern, und postet nur ab und an den gesamten aktuellen Quellcode in einem einzelnen Commit.
+* **Nur eine Hand voll Contributors** => Keine besonders große Entwickler-Community, hoher "Bus-Factor" (Wieviele Leute können vom Bus überfahren werden, ohne dass ein Weiterführen des Projekts gefährdet wäre)
+* **Nur wenige Stars** => Eher unbekanntes Projekt ohne viele "Fans"
+
+Wenn man dann mal schaut, wie viele Repositories nach diesem Schema ein gutes Ranking haben, dann sieht das so aus:
+
+ <iframe width="100%" height="400" scrolling="no" src="https://tomsrocket.github.io/github-publiccode/public/analysis-1.html"></iframe>
+
+Nur ca. ein Viertel der Repositories werden aktiv weiterentwickelt und haben eine aktive Community.
+
+Man könnte nun meinen, das wäre ein schlechter Wert. Aber genauer betrachtet ist bei Quellcode, der von öffentlichen Stellen finanziert wird, ja nicht gefordert, dass dahinter eine aktive Open-Source-Community steht. Vielmehr geht es darum, dass der Code überhaupt als Open Source bereitgestellt wird, und wenn dann sogar 25% davon eine aktive Entwicklercommunity haben, dann ist das ein gutes Ergebnis.
```

Comment:
Weiter oben sprichst du von "Sourcecode", das würde ich vereinheitlichen

Reply:
Also es gab mal eine Zeit, da war das aus SEO-Sicht total sinnvoll, Synonyme im Text zu verwenden.. https://yoast.com/use-synonyms-and-related-keywords/ 
Ich würde das gern so lassen, ich finde das gut.